src/py21cmfast/io/caching.py

```diff
@@ -575,3 +579,18 @@ def noloop(cls):
             halo_field=False,
             xray_source_box=False,
         )
+
+    @classmethod
+    def last_step_only(cls):
+        """Generate a CacheConfig where only boxes needed from more than one step away are cached."""
```

Comment:
can you expand this docstring a bit? I'm not sure I understand the purpose

Reply:
This is the minimum caching setup which allows purging of objects which are needed at multiple redshifts. PerturbedField and PertrubedHaloField, and HaloBox need to be kept in memory at all redshifts if they are not cached, unlike the rest which are generally only needed at the current and last step.